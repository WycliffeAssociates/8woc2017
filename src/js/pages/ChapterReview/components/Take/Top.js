import React from 'react';
import styled from 'styled-components';

export default class Top extends React.Component {

  constructor(props) {
    super(props);
  }


  render() {
    const {takeNum, chunkNum, txt} = this.props;
    return (
      <Container>
        <Text>
          <TakeNum> {txt.get("take")} {takeNum}</TakeNum>

          <ChunkNum> {txt.get("chunk")} {chunkNum}</ChunkNum>
        </Text>

        <Rating> <i className= "material-icons">done </i> </Rating>
      </Container>
    );
  }

}

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 10px;
`;

const Text = styled.div`
`;

const TakeNum = styled.h2`
  font-size: 14px;
`;

const ChunkNum = styled(TakeNum)`
  font-size: 18px;
  margin-top: -10px;
`;
const Rating = styled.div`
`;
