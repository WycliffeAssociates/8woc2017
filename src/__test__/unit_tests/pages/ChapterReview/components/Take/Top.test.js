/* global expect it describe */
import React from 'react';
import renderer from 'react-test-renderer';
import Top from '../../../../../../js/pages/ChapterReview/components/Take/Top';

describe('Take Card Top', () => {
  it('should render the component and not explode', () => {
    const mockProps = {
      txt: {
        take: 'take',
        chunk: 'chunk',
      },
      chunkNum: 1,
      takeNum: 4,
    };
    const tree = renderer
      .create(<Top {...mockProps} />)
      .toJSON();
    expect(tree).toMatchSnapshot();

  });

});
