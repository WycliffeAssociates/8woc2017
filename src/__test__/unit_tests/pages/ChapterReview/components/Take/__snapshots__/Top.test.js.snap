// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take Card Top should render the component and not explode 1`] = `
<div
  className="sc-bdVaJa hoIaPy"
>
  <div
    className="sc-bwzfXH bFHdpU"
  >
    <h2
      className="sc-htpNat buBeEl"
    >
       
       
      4
    </h2>
    <h2
      className="sc-bxivhb knzSls sc-htpNat buBeEl"
    >
       
       
      1
    </h2>
  </div>
  <div
    className="sc-ifAKCX bmKgXx"
  >
     
    <i
      className="material-icons"
    >
      done 
    </i>
     
  </div>
</div>
`;
